src/main/java/com/dumbdogdiner/stickyapi/common/util/textures/TextureHelper.java

```diff
@@ -0,0 +1,239 @@
+/*
+ * Copyright (c) 2020-2021 DumbDogDiner <dumbdogdiner.com>. All rights reserved.
+ * Licensed under the MIT license, see LICENSE for more information...
+ */
+package com.dumbdogdiner.stickyapi.common.util.textures;
+
+import com.dumbdogdiner.stickyapi.StickyAPI;
+import com.dumbdogdiner.stickyapi.common.util.StringUtil;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.reflect.TypeToken;
+import lombok.experimental.UtilityClass;
+import okhttp3.OkHttpClient;
+import org.bukkit.Bukkit;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.yaml.snakeyaml.Yaml;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.lang.reflect.Type;
+import java.net.URL;
+import java.util.*;
+
+/**
+ * Contains various helper methods related to Textures, Texture Strings, and texture validation, as well as providing
+ * access to the bundled texture resources
+ */
+//TODO write javadoc
+@UtilityClass
+public class TextureHelper {
+    // Package-local visibility
+    static final OkHttpClient httpClient = new OkHttpClient();
+    private static final Gson GSON = new Gson();
+    private static final Yaml YAML = new Yaml();
+    private static Map<String, Map<String, String>> TextureMap = generateTextureMap();
```

Comment:
Skeeball really doesn't like C-style variable names - do you use this to identify static vars? 
```suggestion
    private static Map<String, Map<String, String>> textureMap = generateTextureMap();
```

Reply:
Global variables are in C-style (title case), caml case is local, all caps is constant - [CS106a code style](https://youtu.be/GPWah4wbwYs?t=909)

Also trying to avoid this: https://youtu.be/GPWah4wbwYs?t=1264